### PR TITLE
Anko - Change Home layout to Anko

### PR DESCRIPTION
**Pull request by at-hoavo**
#### Related issues (Required)
- Task #109
- Bug #

#### Description (Required)
- [x] Create Home layout 
- [x] Create Item RecyclerView

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                }
+
+                override fun onCollapse(view: ExpandableTextView) {
+                    imgArrowDownBlack.setImageResource(R.drawable.ic_keyboard_arrow_down_black_18dp)
+                }
+            }
+        }
+    }
+
+    inner class HomeAdapterUI : AnkoComponent<ViewGroup> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            }.view
+
+            itemView.tag = HomeViewHolder(itemView, mTvTime, mTvStatus, mImgArrowDownBlack, mExpandableTextView, mImgPoint, mLLItemLocation)
+            return itemView
+        }
+    }
+
+}
+
+inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit): ExpandableTextView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        internal const val ID_ITEM_TIME = 3
+        internal const val ID_ITEM_STATUS = 3
+        internal const val ID_EXPANDABLE_VIEW = 4
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = HomeAdapterUI()
+            .createView(AnkoContext.create(context, parent, false))
+            .tag as HomeViewHolder
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    override fun onBindViewHolder(holder: HomeViewHolder?, position: Int) {
+        holder?.bindHomeViewHolder(locations[position])
+    }
+
+    override fun getItemCount(): Int = locations.size
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt (hunk 2 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+class HomeAdapter(private val context: Context, private val locations: List<TrackingInformation>, val onClickItem: (Int) -> Unit)
+    : RecyclerView.Adapter<HomeAdapter.HomeViewHolder>() {
+
+    companion object {
+        internal const val ID_ITEM_TIME = 3
+        internal const val ID_ITEM_STATUS = 3
+        internal const val ID_EXPANDABLE_VIEW = 4
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = HomeAdapterUI()
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+import org.jetbrains.anko.*
+import org.jetbrains.anko.custom.ankoView
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.TrackingInformation
+
+/**
+ *  Copyright © 2017 AsianTech inc.
+ *  Created by at-hoavo on 27/11/2017.
+ */
+class HomeAdapter(private val context: Context, private val locations: List<TrackingInformation>, val onClickItem: (Int) -> Unit)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+        holder?.bindHomeViewHolder(locations[position])
+    }
+
+    override fun getItemCount(): Int = locations.size
+
+
+    /**
+     * To save data for items in recyclerView of locations
+     */
+    inner class HomeViewHolder(val item: View, val tvTime: TextView, val tvStatus: TextView, val imgArrowDownBlack: ImageView, val expTvDescription: ExpandableTextView, val imgPoint: ImageView, val llItemLocation: LinearLayout) : RecyclerView.ViewHolder(item) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                        }
+
+                        mImgArrowDownBlack = imageView {
+                            visibility = View.GONE
+                            backgroundResource = R.drawable.ic_keyboard_arrow_down_black_18dp
+                        }.lparams {
+                            topOf(HomeAdapter.ID_EXPANDABLE_VIEW)
+                        }
+                    }
+                }
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+            }.view
+
+            itemView.tag = HomeViewHolder(itemView, mTvTime, mTvStatus, mImgArrowDownBlack, mExpandableTextView, mImgPoint, mLLItemLocation)
+            return itemView
+        }
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            mFabMenuGroups = floatingButton {}
+                    .lparams(wrapContent, wrapContent) {
+                        alignParentBottom()
+                        alignParentRight()
+                        margin = dimen(R.dimen.home_screen_floating_button_margin)
+                    }
+        }
+    }
+
+    inline fun ViewManager.floatingButton(init: FloatingButtonHorizontal.() -> Unit): FloatingButtonHorizontal {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING: Public functions require documentation.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        return ankoView({ FloatingButtonHorizontal(it) }, theme = 0, init = init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.TrackingInformation
+import vn.asiantech.way.extension.toast
+import vn.asiantech.way.ui.base.BaseActivity
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+import vn.asiantech.way.ui.register.RegisterActivity
+import vn.asiantech.way.utils.LocationUtil
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+            mPosition = it
+        }
+    }
+
+    private fun initDummyData(locations: ArrayList<TrackingInformation>) {
+//        locations.add(TrackingInformation("1:00 PM", "Stop", "30 minutes| You stop at the school...............",
+//                LatLng(16.0721115, 108.2302225)))
+//        locations.add(TrackingInformation("2:00 PM", "Drive", "50 minutes| You went to the Hoa Khanh market........",
+//                LatLng(16.0712047, 108.2193197)))
+//        locations.add(TrackingInformation("3:00 PM", "Walk", "30 minutes | 1km", LatLng(16.0721611, 108.2303906)))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+}
+
+/**
+ * Function to custom expandableTextView
+ */
+inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit): ExpandableTextView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+                    mLLItemLocation
+            )
+            return itemView
+        }
+    }
+
+}
+
+/**
+ * Function to custom expandableTextView
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+}
+
+/**
+ * Function to custom floatingButton
+ */
+inline fun ViewManager.floatingButton(init: FloatingButtonHorizontal.() -> Unit)
+        : FloatingButtonHorizontal {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+ */
+inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit)
+        : ExpandableTextView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+
+/**
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by atHangTran on 26/09/2017.
+ */
+class HomeActivity : BaseActivity(), OnMapReadyCallback, FloatingButtonHorizontal.OnMenuClickListener {
+
+    companion object {
+        const val PADDING_LEFT = 0
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: please define on `dimen.xml`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+        const val PADDING_TOP = 0
+        const val PADDING_RIGHT = 0
+        private const val UNIT_PADDING_BOTTOM = 3
+    }
+
+    private var mPosition = -1
+    private var mGoogleMap: GoogleMap? = null
+    private lateinit var mHomeAdapter: HomeAdapter
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: please remove `prefix m`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+
+/**
+ * Created at 11 / 2017
+ * Created by at-hoavo on 27/11/2017.
+ */
+class HomeActivityUI(private val homeAdapter: HomeAdapter) : AnkoComponent<HomeActivity> {
+
+    companion object {
+        internal const val ID_MAP = 2
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: Please change id to `1001`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    internal lateinit var mFabMenuGroups: FloatingButtonHorizontal
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: remove `prefix m`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 22]
```diff
@@ -0,0 +1,202 @@
...
+/**
+ *  Copyright © 2017 AsianTech inc.
+ *  Created by at-hoavo on 27/11/2017.
+ */
+class HomeAdapter(private val context: Context,
+                  private val locations: List<TrackingInformation>,
+                  val onClickItem: (Int) -> Unit)
+    : RecyclerView.Adapter<HomeAdapter.HomeViewHolder>() {
+
+    companion object {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: Please correct `id value`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 23]
```diff
@@ -0,0 +1,202 @@
...
+                        imgPoint.setBackgroundResource(R.drawable.ic_point_white)
+                    } else {
+                        itemView.setBackgroundResource(R.drawable.bg_item_location_default)
+                        imgPoint.setBackgroundResource(R.drawable.ic_point_pink)
+                        imgArrowDownBlack.visibility = View.GONE
+                    }
+                }
+            }
+
+            llItemLocation.setOnClickListener {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: use  lambda ` syntax instead for `interface`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 24]
```diff
@@ -0,0 +1,202 @@
...
+                }
+            }
+        }
+    }
+
+    /**
+     * Class to bind ViewHolder with Anko layout
+     */
+    inner class HomeAdapterUI : AnkoComponent<ViewGroup> {
+        internal lateinit var mTvTime: TextView
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: remove prefix m on variables

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 25]
```diff
@@ -0,0 +1,202 @@
...
+                    }
+
+                    relativeLayout {
+                        view {
+                            backgroundColor = ContextCompat.getColor(context, R.color.colorBlack)
+                        }.lparams(dip(1), matchParent) {
+                            centerHorizontally()
+                        }
+
+                        mImgPoint = imageView {}.lparams(
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: change to `imageView()`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 26]
```diff
@@ -0,0 +1,202 @@
...
+                        ) {
+                            centerVertically()
+                        }
+
+                    }.lparams(wrapContent, matchParent) {
+                        leftMargin = dimen(R.dimen.home_screen_view_margin)
+                    }
+
+                    imageView {
+                        backgroundResource = R.drawable.ic_stop
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 27]
```diff
@@ -0,0 +1,202 @@
...
+                        }.lparams(matchParent, wrapContent) {
+                            alignParentLeft()
+                            below(HomeAdapter.ID_ITEM_STATUS)
+                            rightMargin = dimen(R.dimen.home_screen_expTv_marginRight)
+                            topMargin = dimen(R.dimen.home_screen_view_margin)
+                        }
+
+                        mImgArrowDownBlack = imageView {
+                            visibility = View.GONE
+                            backgroundResource = R.drawable.ic_keyboard_arrow_down_black_18dp
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 28]
```diff
@@ -0,0 +1,74 @@
...
+            }
+
+            mFrOverplay = frameLayout {
+                visibility = View.GONE
+            }.lparams(matchParent, matchParent) {
+                backgroundColor = R.color.colorOverlay
+            }
+
+            mFabMenuGroups = floatingButton {}
+                    .lparams(wrapContent, wrapContent) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: remove `wrapContent`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 29]
```diff
@@ -0,0 +1,134 @@
...
+import com.google.android.gms.maps.model.LatLng
+import kotlinx.android.synthetic.main.activity_home.*
+import org.jetbrains.anko.dimen
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.TrackingInformation
+import vn.asiantech.way.ui.base.BaseActivity
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 30]
```diff
@@ -0,0 +1,134 @@
+package vn.asiantech.way.ui.home
+
+import android.graphics.Point
+import android.os.Build
+import android.os.Bundle
+import android.view.WindowManager
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Unused import

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 31]
```diff
@@ -0,0 +1,133 @@
...
+import com.google.android.gms.maps.SupportMapFragment
+import kotlinx.android.synthetic.main.activity_home.*
+import org.jetbrains.anko.dimen
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.TrackingInformation
+import vn.asiantech.way.ui.base.BaseActivity
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 32]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    private fun initDummyData(locations: ArrayList<TrackingInformation>) {
+        // TODO set list WayLocation
+//        locations.add(TrackingInformation("1:00 PM", "Stop", "30 minutes| You stop at the school...............",
+//                LatLng(16.0721115, 108.2302225)))
+//        locations.add(TrackingInformation("2:00 PM", "Drive", "50 minutes| You went to the Hoa Khanh market........",
+//                LatLng(16.0712047, 108.2193197)))
+//        locations.add(TrackingInformation("3:00 PM", "Walk", "30 minutes | 1km", LatLng(16.0721611, 108.2303906)))
+//        locations.add(TrackingInformation("4:00 PM", "Destination", "1 hour ago | 5km", LatLng(16.0725051, 108.2296716)))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+import com.google.android.gms.maps.SupportMapFragment
+import kotlinx.android.synthetic.main.activity_home.*
+import org.jetbrains.anko.dimen
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.TrackingInformation
+import vn.asiantech.way.ui.base.BaseActivity
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 34]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    private fun initDummyData(locations: ArrayList<TrackingInformation>) {
+        // TODO set list WayLocation
+//        locations.add(TrackingInformation("1:00 PM", "Stop", "30 minutes| You stop at the school...............",
+//                LatLng(16.0721115, 108.2302225)))
+//        locations.add(TrackingInformation("2:00 PM", "Drive", "50 minutes| You went to the Hoa Khanh market........",
+//                LatLng(16.0712047, 108.2193197)))
+//        locations.add(TrackingInformation("3:00 PM", "Walk", "30 minutes | 1km", LatLng(16.0721611, 108.2303906)))
+//        locations.add(TrackingInformation("4:00 PM", "Destination", "1 hour ago | 5km", LatLng(16.0725051, 108.2296716)))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 35]
```diff
@@ -0,0 +1,69 @@
...
+
+            fabMenuGroups = floatingButton {}
+                    .lparams {
+                        alignParentBottom()
+                        alignParentRight()
+                        margin = dimen(R.dimen.home_screen_floating_button_margin)
+                    }
+        }
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 36]
```diff
@@ -0,0 +1,192 @@
...
+    }
+
+    override fun getItemCount(): Int = locations.size
+
+    /**
+     * To save data for items in recyclerView of locations
+     */
+    inner class HomeViewHolder(val item: View) : RecyclerView.ViewHolder(item) {
+
+        var tvTime: TextView
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: change this line to `private val tvTime: TextView = item.find(R.id.home_adapter_tv_time)` 
also apply for all variable below

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 37]
```diff
@@ -0,0 +1,192 @@
...
+                            topOf(R.id.home_adapter_tv_expandable_description)
+                        }
+                    }
+                }
+            }.view
+            itemView.tag = HomeViewHolder(itemView)
+            return itemView
+        }
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: remove this line

File: app/src/main/res/values/ids.xml [hunk 38]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Home Screen -->
```
Comment (at-vinhhuynh) [inline app/src/main/res/values/ids.xml]: change comment to `<!-- View ids for HomeAdapter -->`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt (hunk 33 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 39]
```diff
@@ -0,0 +1,137 @@
...
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt [hunk 40]
```diff
@@ -0,0 +1,68 @@
...
+                    }
+        }
+    }
+}
+
+/**
+ * Function to custom floatingButton
+ */
+inline fun ViewManager.floatingButton(init: FloatingButtonHorizontal.() -> Unit)
+        : FloatingButtonHorizontal {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 41]
```diff
@@ -0,0 +1,191 @@
...
+            return itemView
+        }
+    }
+}
+
+/**
+ * Function to custom expandableTextView
+ */
+inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit)
+        : ExpandableTextView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt (hunk 33 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt (hunk 39 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thanks!

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 42]
```diff
@@ -0,0 +1,137 @@
...
+                    recycleViewLocation.scrollToPosition(it - 1)
+                }
+            }
+            locations[it].isChoose = true
+            homeAdapter.notifyItemChanged(it)
+            position = it
+        }
+    }
+
+    private fun initDummyData(locations: ArrayList<TrackingInformation>) {
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: Remove this method

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 43]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    private fun setPaddingGoogleLogo() {
+        val size = Point()
+        windowManager.defaultDisplay.getSize(size)
+        val padding = dimen(R.dimen.padding_logo_map)
+        googleMap?.setPadding(padding, padding, padding, size.y / UNIT_PADDING_BOTTOM)
+    }
+
+    private fun setDataForRecyclerView() {
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: Change this method to `setDataForRecyclerView(locations: ArrayList<TrackingInformation>)`

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt (hunk 33 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 44]
```diff
@@ -0,0 +1,182 @@
...
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt (hunk 40 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 45]
```diff
@@ -0,0 +1,121 @@
...
+import com.google.android.gms.maps.SupportMapFragment
+import kotlinx.android.synthetic.main.activity_home.*
+import org.jetbrains.anko.dimen
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.R
+import vn.asiantech.way.data.model.TrackingInformation
+import vn.asiantech.way.ui.base.BaseActivity
+import vn.asiantech.way.ui.custom.FloatingButtonHorizontal
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt (hunk 40 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivityUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt (hunk 44 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thank you!

**Comment by at-vinhhuynh** [comment]
@hoavothi 
thanks for fixing!
LGTM 👍 

@at-tienhoang 
I finish review this PR, please check it again.
thank you!

**Pull request merged**